lib/src/util/console_log_utils.dart

```diff
@@ -91,26 +99,37 @@ void Function() startSpyingOnConsoleLogs({
   @required void Function(String) onLog,
 }) {
   final logTypeToCapture = configuration.logType == 'all' ? ConsoleConfig.types : [configuration.logType];
-  final consoleRefs = <String, JsFunction>{};
+  final consoleRefs = <String, dynamic>{};
+  final consolePropertyDescriptors = <String, dynamic>{};
 
   _resetPropTypeWarningCache();
 
   // Bind to the current zone so the callback isn't called in the top-level zone.
   final boundOnLog = Zone.current.bindUnaryCallback(onLog);
 
   for (final config in logTypeToCapture) {
-    consoleRefs[config] = context['console'][config] as JsFunction;
-    context['console'][config] = JsFunction.withThis((self, [message, arg1, arg2, arg3, arg4, arg5]) {
-      // NOTE: Using console.log or print within this function will cause an infinite
-      // loop when the logType is set to `log`.
-      boundOnLog(format(message?.toString(), [arg1, arg2, arg3, arg4, arg5]));
-      consoleRefs[config].apply([message, arg1, arg2, arg3, arg4, arg5], thisArg: self);
-    });
+    consolePropertyDescriptors[config] = _getOwnPropertyDescriptor(_console, config);
+    consoleRefs[config] = context['console'][config];
```

Comment:
Could we restore the cast that was previously here along with the typing on `consoleRefs`?